v3/integration/genTestCerts/genTestCerts.go

```diff
@@ -0,0 +1,314 @@
+// more than a signed certificate with IsCA set to true. Not even any
+// basic constraints are defined. Please do not think that this will be
+// acceptable to any system, let alone lint particularly well.
+func NewIntermediate(parent *x509.Certificate) (*x509.Certificate, error) {
+	// Edit this template to look like whatever intermediate you need.
+	template := x509.Certificate{
+		Raw:                         nil,
+		RawTBSCertificate:           nil,
+		RawSubjectPublicKeyInfo:     nil,
+		RawSubject:                  nil,
+		RawIssuer:                   nil,
+		Signature:                   nil,
+		SignatureAlgorithm:          0,
+		PublicKeyAlgorithm:          0,
+		PublicKey:                   nil,
+		Version:                     0,
+		SerialNumber:                NextSerial(),
+		Issuer:                      pkix.Name{},
+		Subject:                     pkix.Name{},
+		NotBefore:                   time.Now(),
+		NotAfter:                    time.Now(),
+		KeyUsage:                    0,
+		Extensions:                  nil,
+		ExtraExtensions:             nil,
+		UnhandledCriticalExtensions: nil,
+		ExtKeyUsage:                 nil,
+		UnknownExtKeyUsage:          nil,
+		BasicConstraintsValid:       false,
+		IsCA:                        true,
+		MaxPathLen:                  0,
+		MaxPathLenZero:              false,
+		SubjectKeyId:                nil,
+		AuthorityKeyId:              nil,
+		OCSPServer:                  nil,
+		IssuingCertificateURL:       nil,
+		DNSNames:                    nil,
+		EmailAddresses:              nil,
+		IPAddresses:                 nil,
+		URIs:                        nil,
+		PermittedDNSDomainsCritical: false,
+		PermittedDNSDomains:         nil,
+		ExcludedDNSDomains:          nil,
+		PermittedIPRanges:           nil,
+		ExcludedIPRanges:            nil,
+		PermittedEmailAddresses:     nil,
+		ExcludedEmailAddresses:      nil,
+		PermittedURIDomains:         nil,
+		ExcludedURIDomains:          nil,
+		CRLDistributionPoints:       nil,
+		PolicyIdentifiers:           nil,
+	}
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+	cert, err := x509.CreateCertificate(rand.Reader, &template, parent, key.Public(), key)
+	if err != nil {
+		return nil, err
+	}
+	return x509.ParseCertificate(cert)
+}
+
+// This is NOT a healthy example of a leaf certificate, this is nothing
+// more than a self signed certificate with IsCA set to false. Not even any
+// basic constraints are defined. Please do not think that this will be
+// acceptable to any system, let alone lint particularly well.
+func NewLeaf(trustAnchor *x509.Certificate, intermediates []*x509.Certificate) (*x509.Certificate, error) {
+	var parent *x509.Certificate
+	if len(intermediates) == 0 {
+		parent = trustAnchor
+	} else {
+		parent = intermediates[len(intermediates)-1]
+	}
+	// Edit this template to look like whatever lead cert you need.
```

Comment:
:warning: Lead certificates are known to the Government Of California to be a carcinogen :warning: 

```suggestion
	// Edit this template to look like whatever leaf cert you need.
```